changelog.d/issue-4683

```diff
@@ -0,0 +1,4 @@
+synopsis: 'cabal check' to fail when no upper bounds for base or Cabal are present in setup dependencies
+packages: cabal-install
```

Comment:
While this affects `cabal-install`, it's also a change in `Cabal`'s `checkPackage`, isn't it?

Reply:
Good point! Fixed (I hope).

Reply:
Thanks!

Reply:
@fgaz do you think you could approve it or is there anything I could do?

Reply:
Sorry, I don't have much time right now, and I really just looked at the changelog when I left that review. @emilypi maybe?

Reply:
No problem, thanks.